perception_dataset/deepen/deepen_annotation.py

```diff
@@ -0,0 +1,222 @@
+            ), f"quaternion is missing keys: {missing_quaternion_keys}"
+
+        def _check_two_d_box() -> None:
+            """
+            Checks if two_d_box has four elements: [corner_x, corner_y, width, height].
+
+            Raises:
+                AssertionError: If the list does not have exactly four numerical elements.
+            """
+            if self.box is None:
+                return
+
+            assert isinstance(self.box, list), "two_d_box must be a list."
+            assert len(self.box) == 4, "two_d_box must be a list of four elements."
+            assert all(
+                isinstance(value, Number) and value >= 0 for value in self.box
+            ), "two_d_box elements must be numbers and greater than 0."
+
+        def _check_two_d_mask() -> None:
+            """
+            Checks if two_d_mask is a non-empty string representing an RLE (Run-Length Encoding).
+            Args:
+                two_d_mask (str): The RLE string to validate.
+            Raises:
+                AssertionError: If two_d_mask is not a valid non-empty string.
+            """
+            if self.two_d_mask is None:
+                return
+
+            assert isinstance(
+                self.two_d_mask["counts"], str
+            ), "two_d_mask['counts'] must be a string."
+            assert self.two_d_mask[
+                "counts"
+            ].strip(), "two_d_mask['counts'] must not be an empty string."
+
+        # Ensures that exactly one of three_d_bbox, two_d_box, or two_d_mask is provided.
+        _check_provided_annotations_exists()
+
+        # Checks if label_id follows the format '{label_category_id}:{int}'
+        _check_label_id()
+
+        # Checks three_d_box keys
+        _check_three_d_bbox()
+
+        # Checks two_d_box keys
+        _check_two_d_box()
+
+        # Checks two_d_mask keys
+        _check_two_d_mask()
+
+    def to_dict(self) -> Dict[str, Any]:
+        """Converts the dataclass instance to a dictionary."""
+        return asdict(self)
+
+    @classmethod
+    def from_file(
+        cls,
+        ann_file: str,
+        *,
+        as_dict: bool = True,
+    ) -> List[DeepenAnnotationLike | Dict[str, Any]]:
+        """Load annotations from file(s).
+
+        Args:
+            ann_file (str): Annotation file (.json).
+            as_dict (bool, optional): Whether to output objects as dict or its instance.
+                Defaults to True.
+
+        Returns:
+            List[DeepenAnnotationLike | Dict[str, Any]]: List of annotations or dicts.
+        """
+        with open(ann_file, "r") as f:
+            data = json.load(f)
+
+        labels: List[Dict[str, Any]] = data["labels"]
+
+        output = []
+        for label in labels:
+            if as_dict:
+                output.append(label)
+            else:
+                output.append(DeepenAnnotation(**label))
+        return output
+
+
+DeepenAnnotationLike = TypeVar("DeepenAnnotationLike", bound=DeepenAnnotation)
```

Comment:
nits: What is the difference between DeepenAnnotationLike and DeepenAnnotation?

Reply:
We can also use `DeepenAnnotation` directly, but then type checker expects only `DeepenAnnotation` instances, not its subclasses such as `DeepenSegmentationPainting2D`.

`TypeVar("...", bound=<BASE_CLASS>`) allows above cases.

FYI:
- https://docs.python.org/ja/3/library/typing.html#typing.TypeVar
- https://stackoverflow.com/questions/68244928/what-does-argument-bound-mean-in-typing-typevar